arch/Kconfig

```diff
@@ -34,6 +34,7 @@ config ARM
 	# is really only necessary for Cortex-M with ARM MPU!
 	select GEN_PRIV_STACKS
 	select ARCH_HAS_THREAD_LOCAL_STORAGE if CPU_AARCH32_CORTEX_R || CPU_CORTEX_M || CPU_AARCH32_CORTEX_A
+	select ARCH_HAS_SW_ISR_TABLE_SHARED_IRQ
```

Comment:
besides the test, there is anything else preventing this to be enabled for other architectures as well?

Reply:
I enabled this on the other architectures and some of the devices fails the shared_isr test.

Reply:
Could we release this feature on a per architecture basis? I'm not sure why this is working for the cortex_m and failing on the other architectures.

Reply:
I mean that's concerning right? We are introducing a new wide features that for some reason is only working one one architecture and failing on all the other ones. At the very minimum we should have an idea of why this is the case.

Anyway, not giving my -1 because I have already blocked this for too long. Not giving +1 either.

Reply:
I agree with you but I don't have the expertise to debug the other architectures. How would you recommend I proceed?